.github/workflows/pylint.yml

```
@@ -0,0 +1,23 @@
+name: Pylint
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.7"]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
+    - name: Analysing the code with pylint
+      run: |
+        pylint $(git ls-files '*.py')
```

Do you think it would be worth changing this so that this only runs on files that were changed in the branch? Otherwise I think it will always just give us the same warnings about files that aren't really relevant to whichever PR its running on.

---

```suggestion
        PYTHONFILES=""
        for file in ${{ steps.changed-files.outputs.all_modified_files }}; do
          if [[ -f $file && $file == *.py ]]
            then
              PYTHONFILES="${PYTHONFILES} $file"
            fi
        done
        echo $PYTHONFILES
        pylint $PYTHONFILES
```

Get output of `changed-files` and filter to match python files